include/psa/crypto_se_driver.h

```diff
@@ -927,7 +927,14 @@ typedef psa_status_t (*psa_drv_se_allocate_key_t)(
  * sake of initial device provisioning or onboarding. Such a mechanism may
  * be added to a future version of the PSA Cryptography API specification.
  *
+ * This function may update the driver's persistent data through
+ * \p persistent_data. The core will save the updated persistent data at the
+ * end of the key creation process. See the description of
+ * ::psa_drv_se_allocate_key_t for more information.
```

Comment:
Should we have a test to make sure that the core saved the persistent data?

Reply:
Yes, but that's part of our tech debt. We don't have one for allocate either.

Reply:
I added some testing (good cases only). And fixed a bug that made it not work on deinit/reinit.